File: app/typescript/src/application/port/in/InventoryUseCase.ts
import { Inventory } from '../../../domain/model/inventory/Inventory';

/**
 * 在庫照会クエリパラメータ
 */
export interface InventoryQuery {
  場所コード?: string;
  品目コード?: string;
}

/**
 * 在庫ユースケース (Input Port)
 */
export interface InventoryUseCase {
  /**
   * 場所・品目別の在庫を照会
   */
  getInventory(locationCode: string, itemCode: string): Promise<Inventory | null>;

  /**
   * 場所別の在庫一覧を照会
   */
  getInventoriesByLocation(locationCode: string): Promise<Inventory[]>;

  /**
   * 品目別の在庫一覧を照会（全場所）
   */
  getInventoriesByItem(itemCode: string): Promise<Inventory[]>;

  /**
   * 在庫一覧を照会（クエリパラメータで絞り込み）
   */
  queryInventories(query: InventoryQuery): Promise<Inventory[]>;
}

File: app/java/src/main/java/com/example/production/application/port/in/InventoryUseCase.java
package com.example.production.application.port.in;

import com.example.production.domain.model.inventory.Stock;

import java.util.List;

/**
 * 在庫ユースケース（Input Port）
 */
public interface InventoryUseCase {

    /**
     * 在庫一覧を取得
     */
    List<Stock> getInventory(InventoryQuery query);

    /**
     * 場所と品目で在庫を取得
     */
    Stock getStock(String locationCode, String itemCode);

    /**
     * 場所別の在庫一覧を取得
     */
    List<Stock> getStocksByLocation(String locationCode);

    /**
     * 品目別の在庫一覧を取得
     */
    List<Stock> getStocksByItem(String itemCode);

    /**
     * すべての在庫を取得
     */
    List<Stock> getAllStocks();

    /**
     * 在庫照会クエリ
     */
    record InventoryQuery(
            String itemCode,
            String locationCode,
            String stockStatus
    ) {}
}
